core/src/collections.rs

```diff
@@ -26,29 +28,107 @@ impl MongoCollections {
     // The query timeout comes from the statement attribute SQL_ATTR_QUERY_TIMEOUT. If there is a
     // timeout, the query must finish before the timeout or an error is returned.
     pub fn list_tables(
```

Comment:
I am sorry, I forgot the table type filter in this signature. That's why `_table_t` in `sql_tables` is still unused 😞

Reply:
I added some handling of the table type that checks for table, view, timeseries, and SQL_ALL_TABLE_TYPES.  Let me know what you think.